src/common/regions/types.ts

```diff
@@ -87,6 +102,21 @@ export class State extends Region {
       (subregion instanceof County && subregion.state === this)
     );
   }
+
+  public toObject(): StateObject {
+    return {
+      n: this.name,
+      u: this.urlSegment,
+      f: this.fipsCode,
+      p: this.population,
+      r: this.regionType,
+      s: this.stateCode,
+    };
+  }
+
+  public static fromObject(obj: StateObject) {
```

Comment:
TIL stringify would take advantage of this.  But because `County` and `MetroArea` still require the lookups to find the referenced regions during parsing, I want to leave it like this to make it consistent between those three types.

In the other diff (https://github.com/covid-projections/covid-projections/pull/3261) I found an approach that I like a little better, which brings the `statesByFips` lookup into the main bundle separately from `countiesByFips` and `metroAreasByFips`, because it's so much smaller and useful as a reference.  To make that work (I think) we need to separate the class definitions into separate files to avoid a circular dependency (e.g, `MetroArea` -> `statesByFips` -> `State`).  Or I suppose we could just bring `statesByFips` definition into that types file.  

There are some tradeoffs with that (loading the lookup data in a `types` file, splitting up class definitions into separate files when having them all together is nice).  There are also some dependencies still I think, getting direct County references out of MetroArea and using CountyFips instead (and doing the lookup where used).

So I see a couple of options:
1. fix the other things in the diff as-is, then make any other changes necessary to eliminate the lookup requirement, then do it by lookup so `toObject` in `County` and `MetroArea` doesn't require a lookup to be passed in in a second PR
2. bring the minimum number of changes necessary to unlock this capability into this PR.

After thinking about the choices, I think at this moment I prefer something like this:
a. bring the "statesByFips" loading and hydration right into `types.ts` so we don't have to split `types.ts` to avoid the circular dependency.  any place that needs `statesByFips` is probably going to need those other types anyway.
b. roll those changes into this PR and land it all as a unit

I'll be working on this stuff tonight, maybe I'll make an example commit with those changes so you can see what I mean, it would at least help me identify any remaining dependencies of that change.